File: Dopamine.InformationModule/InformationModule.cs
using Dopamine.Core.Prism;
using Dopamine.InformationModule.ViewModels;
using Dopamine.InformationModule.Views;
using Microsoft.Practices.Prism.Modularity;
using Microsoft.Practices.Prism.Regions;
using Microsoft.Practices.Unity;

namespace Dopamine.InformationModule
{
    public class InformationModule : IModule
    {
        #region Variables
        private readonly IRegionManager regionManager;
        private IUnityContainer container;
        #endregion

        #region Construction
        public InformationModule(IRegionManager regionManager, IUnityContainer container)
        {
            this.regionManager = regionManager;
            this.container = container;
        }
        #endregion

        #region IModule
        public void Initialize()
        {
            this.container.RegisterType<object, InformationViewModel>(typeof(InformationViewModel).FullName);
            this.container.RegisterType<object, Information>(typeof(Information).FullName);
            this.container.RegisterType<object, InformationAboutViewModel>(typeof(InformationAboutViewModel).FullName);
            this.container.RegisterType<object, InformationAbout>(typeof(InformationAbout).FullName);
            this.container.RegisterType<object, InformationHelpViewModel>(typeof(InformationHelpViewModel).FullName);
            this.container.RegisterType<object, InformationHelp>(typeof(InformationHelp).FullName);
            this.container.RegisterType<object, InformationSubMenu>(typeof(InformationSubMenu).FullName);
            this.container.RegisterType<object, InformationAboutLicense>(typeof(InformationAboutLicense).FullName);

            this.regionManager.RegisterViewWithRegion(RegionNames.InformationRegion, typeof(InformationHelp));
        }
        #endregion
    }
}

File: Dopamine.InformationModule/InformationModule.vb
Imports Microsoft.Practices.Prism.Modularity
Imports Microsoft.Practices.Prism.Regions
Imports Microsoft.Practices.Unity
Imports Dopamine.InformationModule.ViewModels
Imports Dopamine.InformationModule.Views
Imports Dopamine.Core.Prism

Public Class InformationModule
    Implements IModule

#Region "Variables"
    Private ReadOnly mRegionManager As IRegionManager
    Private mContainer As IUnityContainer
#End Region

#Region "Construction"
    Public Sub New(iRegionManager As IRegionManager, iContainer As IUnityContainer)
        Me.mRegionManager = iRegionManager
        Me.mContainer = iContainer
    End Sub
#End Region

#Region "Functions"
    Public Sub Initialize() Implements IModule.Initialize
        Me.mContainer.RegisterType(Of Object, InformationViewModel)(GetType(InformationViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, Information)(GetType(Information).FullName)
        Me.mContainer.RegisterType(Of Object, InformationAboutViewModel)(GetType(InformationAboutViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, InformationAbout)(GetType(InformationAbout).FullName)
        Me.mContainer.RegisterType(Of Object, InformationHelpViewModel)(GetType(InformationHelpViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, InformationHelp)(GetType(InformationHelp).FullName)
        Me.mContainer.RegisterType(Of Object, InformationSubMenu)(GetType(InformationSubMenu).FullName)
        Me.mContainer.RegisterType(Of Object, InformationAboutLicense)(GetType(InformationAboutLicense).FullName)

        Me.mRegionManager.RegisterViewWithRegion(RegionNames.InformationRegion, GetType(Views.InformationHelp))
    End Sub
#End Region
End Class
